### ISSUE TITLE
System.InvalidOperationException

### ISSUE THREAD

**Issue by szogun1910** [NiGHT757/AFKManager#9]
After uploading, the plugin throws errors

```
System.InvalidOperationException: Operations that change non-concurrent collections must have exclusive access. A concurrent update was performed on this collection and corrupted its state. The collection's state is no longer correct.
   at System.ThrowHelper.ThrowInvalidOperationException_ConcurrentOperationsNotSupported()
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.InvalidOperationException: Operations that change non-concurrent collections must have exclusive access. A concurrent update was performed on this collection and corrupted its state. The collection's state is no longer correct.
   at System.Collections.Generic.Dictionary`2.FindValue(TKey key)
   at CounterStrikeSharp.API.Modules.Memory.Schema.GetSchemaOffset(String className, String propertyName) in /home/runner/work/CounterStrikeSharp/CounterStrikeSharp/managed/CounterStrikeSharp.API/Modules/Memory/Schema.cs:line 57
   at CounterStrikeSharp.API.Modules.Memory.Schema.GetPointer[T](IntPtr pointer, String className, String memberName) in /home/runner/work/CounterStrikeSharp/CounterStrikeSharp/managed/CounterStrikeSharp.API/Modules/Memory/Schema.cs:line 114
   at CounterStrikeSharp.API.Core.CEntityIdentity.get_Next() in /home/runner/work/CounterStrikeSharp/CounterStrikeSharp/managed/CounterStrikeSharp.API/Core/Objects.g.cs:line 9413
   at CounterStrikeSharp.API.Utilities.FindAllEntitiesByDesignerName[T](String designerName)+MoveNext() in /home/runner/work/CounterStrikeSharp/CounterStrikeSharp/managed/CounterStrikeSharp.API/Utilities.cs:line 64
   at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Boolean& found)
   at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source)
   at AFKManager.AFKManager.AfkTimer_Callback() in D:\servers\lua\AFKManager\AFKManager\AFKManager.cs:line 168
   at InvokeStub_Action.Invoke(Object, Object, IntPtr*)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
```

**Comment by NiGHT757** [comment]
> After uploading, the plugin throws errors
> 
> ```
> System.InvalidOperationException: Operations that change non-concurrent collections must have exclusive access. A concurrent update was performed on this collection and corrupted its state. The collection's state is no longer correct.
>    at System.ThrowHelper.ThrowInvalidOperationException_ConcurrentOperationsNotSupported()
> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
>  ---> System.InvalidOperationException: Operations that change non-concurrent collections must have exclusive access. A concurrent update was performed on this collection and corrupted its state. The collection's state is no longer correct.
>    at System.Collections.Generic.Dictionary`2.FindValue(TKey key)
>    at CounterStrikeSharp.API.Modules.Memory.Schema.GetSchemaOffset(String className, String propertyName) in /home/runner/work/CounterStrikeSharp/CounterStrikeSharp/managed/CounterStrikeSharp.API/Modules/Memory/Schema.cs:line 57
>    at CounterStrikeSharp.API.Modules.Memory.Schema.GetPointer[T](IntPtr pointer, String className, String memberName) in /home/runner/work/CounterStrikeSharp/CounterStrikeSharp/managed/CounterStrikeSharp.API/Modules/Memory/Schema.cs:line 114
>    at CounterStrikeSharp.API.Core.CEntityIdentity.get_Next() in /home/runner/work/CounterStrikeSharp/CounterStrikeSharp/managed/CounterStrikeSharp.API/Core/Objects.g.cs:line 9413
>    at CounterStrikeSharp.API.Utilities.FindAllEntitiesByDesignerName[T](String designerName)+MoveNext() in /home/runner/work/CounterStrikeSharp/CounterStrikeSharp/managed/CounterStrikeSharp.API/Utilities.cs:line 64
>    at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Boolean& found)
>    at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source)
>    at AFKManager.AFKManager.AfkTimer_Callback() in D:\servers\lua\AFKManager\AFKManager\AFKManager.cs:line 168
>    at InvokeStub_Action.Invoke(Object, Object, IntPtr*)
>    at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
>    --- End of inner exception stack trace ---
> ```

What version of CSS do you have?

**Comment by szogun1910** [comment]
v57

**Comment by NiGHT757** [comment]
> v57

Sorry for late reply, i never had this error in these days.
Do you still have this error?

**Comment by mayersgamer** [comment]
hey bro did you fix why the plugin ignoring the timmer ? sometimes the players wont get kicked to spec and timer not working

**Comment by mayersgamer** [comment]
yes

**Comment by NiGHT757** [comment]
[AFKManager.zip](https://github.com/NiGHT757/AFKManager/files/13489281/AFKManager.zip)


> yes

**Comment by mayersgamer** [comment]
serisouly bro server is crashing every time it loads

**Comment by NiGHT757** [comment]
Please add me on discord

7.night

**Comment by szogun1910** [comment]
@NiGHT757 You can send me also, I will check if it works properly

**Comment by mayersgamer** [comment]
the more players are joining the ignoring start , if i was only alone on the server or with another player it kicks me to spec , but the more they join the random it becames

**Comment by NiGHT757** [comment]
These problems should be fixed in [v0.0.9](https://github.com/NiGHT757/AFKManager/releases/tag/v0.0.9).
